### ISSUE TITLE
bug in tutorials?

### ISSUE THREAD

**Issue by jsay** [ScPoEcon/ScPoEconometrics#54]
On R version 3.4.4 (2018-03-15)
I've loaded the libraries ScPoEconometrics 0.0.1 and learner 0.9.2.1.

When running:
> run_tutorial("chapter3",package="ScPoEconometrics")
I obtain:
Listening on http://127.0.0.1:7177
Warning: Error in value[[3L]]: Couldn't normalize path in `addResourcePath`, with arguments: `prefix` = 'font-awesome-4.5.0'; `directoryPath` = '' [No stack trace available]

Related to the following issue:
https://stackoverflow.com/questions/51213726/when-running-a-tutorial-with-learnr-r-gives-an-error

**Comment by floswald** [comment]
Thanks that’s a very valuable report. Can you post “sessionInfo()” after you see that bug on your computer please? The SO post is without solution unfortunately.

**Comment by jsay** [comment]
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.1 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8    
 [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8   
 [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] distr_2.7.0            sfsmisc_1.1-2          startupmsg_0.9.5      
 [4] EnvStats_2.3.1         learnr_0.9.2.1         bindrcpp_0.2.2        
 [7] plotly_4.8.0           ggplot2_3.0.0          dplyr_0.7.6           
[10] shiny_1.1.0            ScPoEconometrics_0.0.1 devtools_1.13.6       

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4   purrr_0.2.5        colorspace_1.3-2   htmltools_0.3.6   
 [5] viridisLite_0.3.0  yaml_2.2.0         rlang_0.2.1        later_0.7.3       
 [9] pillar_1.3.0       glue_1.3.0         withr_2.1.2        RColorBrewer_1.1-2
[13] bindr_0.1.1        plyr_1.8.4         stringr_1.3.1      munsell_0.5.0     
[17] gtable_0.2.0       htmlwidgets_1.2    memoise_1.1.0      evaluate_0.11     
[21] knitr_1.20         httpuv_1.4.5       crosstalk_1.0.0    curl_3.2          
[25] markdown_0.8       Rcpp_0.12.18       xtable_1.8-2       promises_1.0.1    
[29] scales_0.5.0       backports_1.1.2    jsonlite_1.5       mime_0.5          
[33] digest_0.6.15      stringi_1.2.4      grid_3.4.4         rprojroot_1.3-2   
[37] tools_3.4.4        magrittr_1.5       lazyeval_0.2.1     tibble_1.4.2      
[41] crayon_1.3.4       tidyr_0.8.1        pkgconfig_2.0.1    MASS_7.3-49       
[45] data.table_1.11.4  assertthat_0.2.0   rmarkdown_1.10     httr_1.3.1        
[49] R6_2.2.2           git2r_0.23.0       compiler_3.4.4

**Comment by jsay** [comment]
I've updated (now R version 3.5.1 (2018-07-02)), and the message error change to:

 Error in shiny_prerendered_html(input_rmd, encoding, render_args) : 
  Prerendered HTML file not found at ./chapter3.html

**Comment by floswald** [comment]
In fact, another solution would be to have the students call

rmarkdown::run(filename.rmd),

But that filename would be something like
system.file(“ScPoEconometrics”,”tutorials”,”chapter3”,”chapter3.rmd”)

A bit unwieldy! No?

**Comment by jsay** [comment]
Yes,
I do not know sufficiently the learnr package and its interaction with rmarkdown. 

> rmarkdown::run(ScPoEconometrics/inst/tutorials/chapter3/chapter3.Rmd)
returns:
> Erreur : pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).

This needs additional dependencies... I would favor solutions that work on R out of the box (at least for students)

**Comment by floswald** [comment]
Yes. The tutorials are still WIP. don’t despair, we’ll get there.

It will be good to test the tutorials in the coming days as I have not
found a way to test them via CI.

**Comment by floswald** [comment]
What’s your rstudio build? Pandoc ships as a component of rstudio. Maybe an
update is enough?

**Comment by floswald** [comment]
Ok I confirm that. If you call that function from rstudio you should be
fine (recent version). I’ll tighten up the rmarkdown version requirement in
the next update.
It will be recommended to use R mainly through Rstudio.

**Comment by jsay** [comment]
I do not use Rstudio usually, I'll try it

**Comment by floswald** [comment]
yes if you could, please give it a try. we'll recommend students to use it, but it's important to know that there are version requirements outside of Rstudio that we need to be aware of. I'll try to fix this today, do you have time for a couple of tests later on? thanks

**Comment by floswald** [comment]
ok, that is good to know, thanks! I would still like to be able to use the function `run_tutorial` instead. I'll give it another try tomorrow with recompiling all the tutorials. maybe you could try again then, I'll ping you on slack.

**Comment by floswald** [comment]
pinging @jsay to try this out again. Upon reinstallation you will find function `runTutorial()` in the package which should render everything on your computer (always RStudio!). cheers
